### ISSUE TITLE
Problem protobuf

### ISSUE THREAD

**Issue by josels-09** [fabiocarrara/deep-parking#12]
Great job Fabio, I'm trying to execute the experiments but I've had difficulties.

I currently have the following problem:
**AttributeError: 'google.protobuf.pyext._message.RepeatedScalarConta' object has no attribute '_values'**

What does not allow me to go to training, I would like to know which version of protobuf is used for the experiments, and also the version of python, or if you know how to correct the error.

Thank you.

**Comment by fabiocarrara** [comment]
Hi @josels-09,
unfortunately, I do not have the environment on which I run the experiment anymore, and I don't remember the exact version numbers. I'm quite sure I was using Caffe with Python 2 and protobuf 2.x, probably 2.6.x, surely not 3.x. For the Caffe version, I think you can pull a commit with the same age as this repo and you will get close enough to the version I used.

I'm sorry but I didn't experience this error before, can you share which line of which script raises it?

**Comment by josels-09** [comment]
Thanks for your reply, I could solve the problem, it was related to the versions of the python dependencies.

Now I have another doubt, when I execute the pre-trained models I get a numpy array, which only shows me two values [0. 1.], I already made several attempts and I do not know how to interpret the result, I understand that it should be a probability (free, busy), but I do not know if a conversion is necessary, thanks again.

**Comment by fabiocarrara** [comment]
Yes, the output for an image is a numpy array with two values being the probabilities for free/busy classes, no conversions needed.
If the model gives you always the same output, please check that your data preprocessing pipeline is correct and coherent with the one we used when training our models. See #6 for more info.

**Comment by josels-09** [comment]
Does this mean that I can not test the pre-trained models with the training patches?

**Comment by nikola310** [comment]
@josels-09 Did you find the solution for testing images? I've been trying to use patched images, but no luck so far. From what I understand image we feed to network via forward.py script transformer should correctly preprocess and resize it. Still, the only results so far are [0. 1.]

Is it possible to get any help on this issue @fabiocarrara ?

**Comment by josels-09** [comment]
@nikola310 Yes, I found the solution, the images must be previously converted to the size that is specified, if I'm not wrong 256x256 in BGR format, this can be done with openCV, as they say in the publication of the previous answer, it works for me.

Link:
https://github.com/fabiocarrara/deep-parking/issues/6

Regards.
